packages/amplify-codegen/src/codegen-config/index.js

```diff
@@ -1,12 +1,69 @@
 const AmplifyCodeGenConfig = require('./AmplifyCodeGenConfig');
+const fs = require('fs-extra');
+const path = require('path');
 
 let config = null;
 
 function loadConfig(context, withoutInit = false) {
   if (!config) {
-    config = new AmplifyCodeGenConfig(context, withoutInit);
+    const projectPath = context.amplify.getEnvInfo().projectPath;
+    config = new AmplifyCodeGenConfig(projectPath, withoutInit);
   }
   return config;
 }
 
-module.exports = loadConfig;
+function getCodegenConfig(projectPath) {
+  if (!projectPath) {
+    throw new Error('Invalid projectPath provided to get codegen configuration');
```

Comment:
nit: The `project path` can be used instead of `projectPath` for the error string part. Same as below.